diagram-viewer/src/main/java/com/powsybl/diagram/viewer/MainViewController.java

```diff
@@ -152,6 +159,25 @@ public TreeItem<Container<?>> fromString(String string) {
         sldViewController.addListener((observable, oldValue, newValue) -> updateSldDiagrams());
     }
 
+    private void initializeNetworkFactories() {
```

Comment:
I think we should avoid using reflections to list the test factories, it's too hacky to do it that way. This could easily be done at the cost of listing by hand the available NetworkFactories together with their create methods (`Supplier<Network>`). Later we could add a ServiceLoader if we do the corresponding changes in powsybl-core.